src/PureScript/Backend/Optimizer/CoreFn/Json.purs

```diff
@@ -279,6 +281,15 @@ decodeLiteral dec json = do
     _ ->
       throwError $ TypeMismatch "Literal"
 
+decodeStringFromStringOrFromCodePointArray :: Json -> JsonDecode String
+decodeStringFromStringOrFromCodePointArray json = decodeString json <|> \_ -> map fromCodePointArray $ decodeArray decodeCodePoint json
+
+decodeStringLiteral :: Json -> JsonDecode String
+decodeStringLiteral json =
+  case decodeStringFromStringOrFromCodePointArray json of
+    Right x -> Right x
+    Left error -> Left $ Named "StringLiteral" error
```

Comment:
As I understand it: the compiler will emit a codepoint array when a string literal is invalid utf, such as a lone surrogate.

In general, this code isn't using `Named` annotations, and just opts for catch call `TypeMismatch` errors. I prefer this as an error because with `Either` error propagation and `alt`, you will only see the last error on failure anyway. So in the case of a StringLiteral failing to parse, you will always see `Named "StringLiteral" (insert codepoint array error)`, which doesn't really add anything beyond just `TypeMismatch "StringLiteral"`.

```purescript
decodeStringLiteral json =
  decodeString json <|> \_ -> decodeCodePointArray json <|> \_ -> throwError (TypeMismatch "StringLiteral")
decodeString = ...
decodeCodePointArray = ...
decodeCodePoint = ...
```